notification/endpoint/webhook.go

```diff
@@ -0,0 +1,154 @@
+package endpoint
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/influxdata/influxdb"
+)
+
+var _ influxdb.NotificationEndpoint = &WebHook{}
+
+const (
+	webhookTokenSuffix    = "-token"
+	webhookUsernameSuffix = "-username"
+	webhookPasswordSuffix = "-password"
+)
+
+// WebHook is the notification endpoint config of webhook.
+type WebHook struct {
+	Base
+	// Path is the API path of WebHook
+	URL string `json:"url"`
+	// Token is the bearer token for authorization
+	Token           influxdb.SecretField `json:"token,omitempty"`
+	Username        influxdb.SecretField `json:"username,omitempty"`
+	Password        influxdb.SecretField `json:"password,omitempty"`
+	AuthMethod      string               `json:"authmethod"`
+	Method          string               `json:"method"`
+	ContentTemplate string               `json:"contentTemplate"`
+}
+
+// BackfillSecretKeys fill back fill the secret field key during the unmarshalling
+// if value of that secret field is not nil.
+func (s *WebHook) BackfillSecretKeys() {
+	if s.Token.Key == "" && s.Token.Value != nil {
+		s.Token.Key = s.ID.String() + webhookTokenSuffix
+	}
+	if s.Username.Key == "" && s.Username.Value != nil {
+		s.Username.Key = s.ID.String() + webhookUsernameSuffix
+	}
+	if s.Password.Key == "" && s.Password.Value != nil {
+		s.Password.Key = s.ID.String() + webhookPasswordSuffix
+	}
+}
+
+// SecretFields return available secret fields.
+func (s WebHook) SecretFields() []influxdb.SecretField {
+	arr := make([]influxdb.SecretField, 0)
+	if s.Token.Key != "" {
+		arr = append(arr, s.Token)
+	}
+	if s.Username.Key != "" {
+		arr = append(arr, s.Username)
+	}
+	if s.Password.Key != "" {
+		arr = append(arr, s.Password)
+	}
+	return arr
+}
+
+var goodWebHookAuthMethod = map[string]bool{
```

Comment:
👍 for validating enums. This might be easier to read if the return value for these keys was `true`, and this:

`if _, ok := goodWebHookAuthMethod[s.AuthMethod]; !ok {`
was changed to this:
'if goodWebHookAuthMethod[s.AuthMethod] {`.

Same for other places we're doing this.